tests/integration/fns.sh

```diff
@@ -163,7 +163,22 @@ launch_evm() {
   fi
 }
 
-metallica
+ensure_process_count_equals() {
+  local process_name="$1"
+  local expected_count="$2"
+  local actual_count=$(pgrep -f "$process_name" | wc -l)
+  [ "$actual_count" -eq "$expected_count" ]
+  return $?
+}
+
+gracefull_shutdown() {
+  pkill -15 -f "target/debug/enclave" || true
+  sleep 10
+  ensure_process_count_equals "target/debug/enclave" 0 || return 1
+  kill_em_all
+}
+
+kill_em_all
```

Comment:
is this duplicated since it's called on `gracefull_shutdown()` or?

Reply:
No this runs before every test. Will add a comment.

Reply:
makes sense, thx for the quick explanation!